src/Strategy/Conda.hs

```diff
@@ -0,0 +1,57 @@
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Strategy.Conda (
+  discover,
+) where
+
+import Discovery.Walk
+import Effect.Exec
+import Effect.ReadFS
+import Control.Carrier.Diagnostics hiding (fromMaybe)
+import Path
+import Types
+import Graphing (Graphing)
+import qualified Strategy.Conda.CondaList as CondaList
+import qualified Strategy.Conda.EnvironmentYml as EnvironmentYml
+
+discover :: (Has ReadFS sig m, Has Diagnostics sig m, Has ReadFS rsig run, Has Exec rsig run, Has Diagnostics rsig run) => Path Abs Dir -> m [DiscoveredProject run]
+discover dir = map mkProject <$> findProjects dir
+
+findProjects :: (Has ReadFS sig m, Has Diagnostics sig m) => Path Abs Dir -> m [CondaProject]
+findProjects = walk' $ \dir _ files -> do
+  case findFileNamed "environment.yml" files of
+    Nothing -> pure([], WalkContinue)
+    Just envYml -> do
+      let project =
+            CondaProject
+              {
+                condaDir = dir,
+                condaEnvironmentYml = envYml
+              }
+      pure([project], WalkSkipAll) -- Once we find an environment.yml file, skip the rest of the walk
+
+data CondaProject =
+    CondaProject
+      { condaDir :: Path Abs Dir,
+        condaEnvironmentYml :: Path Abs File
+      } deriving (Eq, Ord, Show)
+
+mkProject :: (Has ReadFS sig n, Has Exec sig n, Has Diagnostics sig n) => CondaProject -> DiscoveredProject n
+mkProject project =
+  DiscoveredProject
+    { projectType = "conda",
+      projectBuildTargets = mempty,
+      projectDependencyGraph = const $ getDeps project, -- what does const do here?
```

Comment:
TLDR: `const` here represents that we don't care about or cannot filter on subprojects.

### Long version
For tools like `gradle`, subprojects can be hugely important, and scanning them all at once is simply not a one-size-fits-all solution.  We have to use `BuildTarget`s to signal that we only care about certain subprojects, and if a strategy can honor that, it will do that by using a function other than `const`.  If we cared about subprojects we could define `getSpecificDeps :: FooProject -> Set BuildTarget -> Graphing Dependency`, and then use `getSpecificDeps project` instead of `const $ getDeps project`

`const :: a -> b -> a`, when applied with only one argument, creates a function that ignores it's argument and returns the first applied argument.

```haskell
const :: a -> b -> a
const a _ = a

const 3 4 = 3
const 4 3 = 4
const "hello" (Just "darkness") = "hello"

Reply:
that makes sense, good to know